### PR TITLE
Improve MVP in ReleaseScreenPopup

### PR DESCRIPTION
**Pull request by yurloc**
* Increased code coverage
* Improved i18n
* Improved release dialog validation
* ReleaseScreenPopupViewImpl no longer leaks into RepositoryStructure

### CODE REVIEW

**Comment by manstis** [comment]
@wmedvede Would you mind taking a look at this please?

**Comment by csadilek** [comment]
+1 esp. for the increased coverage but waiting for @wmedvede to confirm. 

This PR should target master though (not 6.3.x).

File: guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/main/java/org/guvnor/asset/management/client/editors/repository/structure/release/ReleaseScreenPopupViewImpl.java [hunk 1]
```diff
@@ -189,28 +185,31 @@ public String trim( String value ) {
     }
 
     @Override
-    public void setVersionStatus( ControlGroupType status ) {
-        versionTextGroup.setType( status );
+    public void showErrorVersionEmpty() {
+        versionTextGroup.setType( ControlGroupType.ERROR );
+        versionTextHelpInline.setText( Constants.INSTANCE.FieldMandatory0( "Version" ) );
```
Comment (Rikkola) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/main/java/org/guvnor/asset/management/client/editors/repository/structure/release/ReleaseScreenPopupViewImpl.java]: "Version" here is not i18n. I do see you fixed the others.

Comment (yurloc) [inline guvnor-asset-mgmt/guvnor-asset-mgmt-client/src/main/java/org/guvnor/asset/management/client/editors/repository/structure/release/ReleaseScreenPopupViewImpl.java]: Good catch, thanks.

**Comment by yurloc** [comment]
This was meant to be a follow-up to #178. I understand it's not going to land to 6.3.x. Feel free to add more comments and/or close.

I'll open a new PR to master, possibly incorporating #178.

**Comment by porcelli** [comment]
closing as 6.3.x is only blockers

**Pull request closed without merging**